pkgs/build-support/go/module.nix

```diff
@@ -155,14 +141,9 @@ let
   }; in modArgs // (
       {
         outputHashMode = "recursive";
-      } // (if (vendorHashType == "sha256") then {
-        outputHashAlgo = "sha256";
-        outputHash = vendorSha256;
-      } else {
         outputHash = vendorHash;
-      }) // (lib.optionalAttrs (vendorHashType == "sri" && vendorHash == "") {
-        outputHashAlgo = "sha256";
-      })
+        outputHashAlgo = if args' ? vendorSha256 || vendorHash == "" then "sha256" else null;
```

Comment:
It isn't allowed in derivations, it's a convenience for generating the `go-modules` hash otherwise it errors with an empty hash:

```
error: empty hash requires explicit hash type
```

Currently:
```
warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
these 2 derivations will be built:
  /nix/store/f9lqyz4nlzg4smd3qz45dg1vkxk8qzi0-fzf-0.40.0-go-modules.drv
  /nix/store/3lpry8y4535gdhbvf6y78na1qphf5857-fzf-0.40.0.drv
building '/nix/store/f9lqyz4nlzg4smd3qz45dg1vkxk8qzi0-fzf-0.40.0-go-modules.drv'...
unpacking sources
unpacking source archive /nix/store/ph5jpqrhd05y17b94ig0pq3574f7c9ay-source
source root is source
patching sources
configuring
building
go: downloading golang.org/x/sys v0.7.0
go: downloading github.com/mattn/go-isatty v0.0.17
go: downloading github.com/mattn/go-runewidth v0.0.14
go: downloading github.com/rivo/uniseg v0.4.4
go: downloading github.com/gdamore/tcell/v2 v2.5.4
go: downloading github.com/mattn/go-shellwords v1.0.12
go: downloading github.com/saracen/walker v0.1.3
go: downloading golang.org/x/term v0.7.0
go: downloading golang.org/x/sync v0.0.0-20220722155255-886fb9371eb4
go: downloading golang.org/x/text v0.5.0
go: downloading github.com/gdamore/encoding v1.0.0
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
installing
error: hash mismatch in fixed-output derivation '/nix/store/f9lqyz4nlzg4smd3qz45dg1vkxk8qzi0-fzf-0.40.0-go-modules.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-SSz4oHUgfMRbvpdIl1xepfckef1HDA1y646FWnyBp6o=
error: 1 dependencies of derivation '/nix/store/3lpry8y4535gdhbvf6y78na1qphf5857-fzf-0.40.0.drv' failed to build
```
Does that answer your question or have I misunderstood what you were asking?

Reply:
I see. That makes sense.